internal/backend/auth/authz/input.go

```diff
@@ -73,56 +113,27 @@ func buildInput(ctx context.Context, db db.DB, id sdktypes.ID, action string, cf
 			continue
 		}
 
-		oid, err := db.GetOrgIDOf(ctx, id)
-		if err != nil {
-			return nil, fmt.Errorf("get project org: %w", err)
-		}
-
-		associations[name] = map[string]any{
-			"id": id.String(),
-		}
-
-		if oid.IsValid() {
-			oidsSet[oid.String()] = true
-			associations[name]["org_id"] = oid.String()
-		}
-
-		if id.Kind() == sdktypes.ProjectIDKind {
-			pidsSet[id.String()] = true
-			associations[name]["project_id"] = id.String()
-		} else {
-			pid, err := db.GetProjectIDOf(ctx, id)
-			if err != nil {
-				return nil, fmt.Errorf("get project id: %w", err)
-			}
-
-			if pid.IsValid() {
-				pidsSet[pid.String()] = true
-				associations[name]["project_id"] = pid.String()
-			}
+		if associations[name], err = hydrate(ctx, db, id, nil); err != nil {
+			return nil, err
 		}
 	}
 
-	data := map[string]any{
-		// Authenticated user (requester) information.
-		"authn_user":      authnUser,               // requester authn user.
-		"authn_user_id":   authnUser.ID().String(), // requester authn user id.
-		"authn_user_orgs": membershipsMap,          // org memberships related to the authn user: {org_id: {"status": "INVITED", "roles": [str]}}
-
-		// Resource information.
-		"kind":                id.Kind(),    // resource kind. available even if id is invalid, as it still contains the kind.
-		"action_type":         actType,      // [type:]xxx of action, or "" if not specified.
-		"action":              act,          // [xxx:]action of action.
-		"resource_id":         id.String(),  // resource id.
-		"resource_org_id":     oid.String(), // resource org id, if resource id is valid. else "".
-		"resource_project_id": pid.String(), // resource project id, if resource id is valid. else "".
-		"data":                cfg.data,     // aux data supplied by the caller.
-
-		// Explicit associations given as part of the check call.
-		"associated_org_ids":     slices.Collect(maps.Keys(oidsSet)), // all unique non-zero associated org ids and the resource org id.
-		"associated_project_ids": slices.Collect(maps.Keys(pidsSet)), // all unique non-zero associated project ids and the resource project id.
-		"associations":           associations,                       // name -> {"org_id": "xxx"} of associated resources.
+	// TODO: Use https://docs.styra.com/opa/rego-by-example/builtins/regex/globs_match to match permissions.
```

Comment:
Linear ID, at least to investigate this. This is to document your current understanding of this task in detail, and known context - both of which you'll forget in a couple of days, and this comment will stay here forever.